config.go

```diff
@@ -144,15 +172,19 @@ func stringToSlice(s string) []string {
 // convertAndSetSlice builds a slice of a dynamic type.
 // It converts each entry in "values" to the elemType of the passed in slice.
 // The slice remains nil if "values" is empty.
-func convertAndSetSlice(slicePtr interface{}, values []string) {
+func convertAndSetSlice(slicePtr interface{}, values []string) error {
 	sliceVal := reflect.ValueOf(slicePtr).Elem()
 	elemType := sliceVal.Type().Elem()
 
 	for _, s := range values {
 		valuePtr := reflect.New(elemType)
-		convertAndSetValue(valuePtr.Interface(), s)
+		if err := convertAndSetValue(valuePtr.Interface(), s); err != nil {
+			return err
```

Comment:
returning error here, rather than aggregating into some error type, will cause slice conversion to exit early.

Example:
`convertAndSetSlice([]int{}, []string{"1", "2", "c", "4"})`. The int slice will contain the values `1,2`, but not `4`.

Reply:
Yeh I set it up to return on first error.

This could be worth aggregating errors though, nothing more annoying than fixing one line of config only to have the next one fail.

Really it should be setting the values back to 'zero'

Reply:
So actually, would it be better to set the slice back to it's original value (ie nil) and return error. Or attempt to process every value, leaving the slice in an inconsistent state and return a list of errors?

Reply:
My vote is to keep it partially filled. The consumer should be validating for correctness by both inspecting the config and the returned error(s).

Our error can make it clear which fields and indexes were malformed. They can then handle it as they wish.